.github/workflows/tests.yaml

```diff
@@ -1,18 +1,93 @@
 name: Tests
 on: [push, pull_request]
 jobs:
-  test:
+  test-linux:
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, windows-latest]
-    runs-on: ${{ matrix.os }}
+        target:
+        - linux-amd64-unit-test-2-cpu
+        - linux-amd64-unit-test-4-cpu
+        - linux-amd64-unit-test-4-cpu-race
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
     - uses: actions/setup-go@v3
       with:
         go-version: "1.17.13"
     - run: make fmt
-    - run: make race
-    - run: make test
+    - env:
+        TARGET: ${{ matrix.target }}
+      run: |
+        case "${TARGET}" in
+          linux-amd64-unit-test-2-cpu)
+            CPU=2 make test-simulate
+            CPU=2 make test
+            ;;
+          linux-amd64-unit-test-4-cpu)
+            CPU=4 make test-simulate
+            CPU=4 make test
+            ;;
+          linux-amd64-unit-test-4-cpu-race)
+            CPU=4 ENABLE_RACE=true make test-simulate
+            CPU=4 ENABLE_RACE=true make test
```

Comment:
No idea about the cpu usage when enable `-race`.

It seems that the standard github action runner doesn't allow user to run high cpu usage even if it provides two vcore to us. It will shutdown the runner or kill the process with SIGTERM. 

* https://github.com/actions/runner-images/issues/6680#issuecomment-1335778010
* https://github.com/actions/runner-images/issues/6709

I didn't see it before. maybe we can enable `-race` later?

@ahrtr 

REF:
* https://github.com/fuweid/bbolt/actions/runs/3827981238/jobs/6513066220 (runner sends SIGTERM to `go test`
* https://github.com/fuweid/bbolt/actions/runs/3828126044/jobs/6513369814 (shutdown the runner)
* https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
Etcd is running unit tests with 'race': https://github.com/etcd-io/etcd/actions/runs/3821152800/jobs/6500015229

I would try... 
If it would not work, I would try running the `--short`tests alone (that should be quick enough to not trigger high CPU usage).

Reply:
Yeah. I was trying to use new go version, use `taskset -c 0` to limit it in one core and limit it with one go queue. 
But it all failed. Maybe it hits some limitation in the underlay of runner...

Reply:
My mental model is that we should be deprecating array-freelist... 
I'm not aware about adventages over the map-driven implementation. 
So I would be OK for not testing in --race the array one.